documents/final-submission-document/chapters/5_discussion_and_conclusion.tex

```diff
@@ -23,28 +23,100 @@ \section{Achieved Result}
+We think that this approach led to a high overall documentation standard.
+
+It was difficult to summarise the technical background and describe our architecture due to the different terminology used by the XMPP and IETF standards.
+We discuss this in Section~\ref{lessons-learned-standards}.
+
+\subsection{Decisions}
+\subsubsection{Architecture}
+
+Architecture-relevant decisions were carried out and justified in the form of architectural design decisions \cite{architectural-design-decisions} (see Appendix~\ref{sec:architectural-decisions}).
+
+This approach helped us to systematically document influences and plan the architecture in a structured way.
+
+Retrospectively, we should have created more architectural decisions later on in the project, e.g. concerning barrel imports or to establish layering guidelines.
+
+\subsection{Development, Frameworks and Tooling}
+
+\paragraph{Test Driven Development} was not possible the way we had anticipated it.
+
+Due to the use of Angular and stanza, the testing environment differed substantially from the actual application context.
+Therefore, it was challenging to create tests before implementing most of the actual code structure.
+A factor that also contributed to this difficulty was our lack of prior knowledge of the used frameworks.
+
+Nevertheless, writing many tests proved to be very valuable during the project.
+It helped us be confident during development and will be useful to future developers expanding the application.
+
+\paragraph{The Docker Development Environment} has proven itself to be valuable for a quick start in development.
+It provides developers with a very productive way to test modifications in a realistic yet portable environment.
+
+\paragraph{Compodoc,} the tool we used to document and visualise the structure of our Angular application,
+did not add as much additional value to the project documentation as we had hoped.
+
+We assume that Compodoc is be better suited for Angular libraries than applications.
+
+\subsection{Standards}\label{lessons-learned-standards}
+
+During the course of this thesis, we learned several lessons about working with standards and in which way these standards posed challenges or supported our work.
+
+\subsubsection{Terminology}
+
+The \gls{xmpp-grid} standard uses \gls{sacm} terminology \cite{ietf-sacm-terminology-14}, whereas the \gls{xmpp} standard and all \glspl{xep} use a different terminology.
+Most concepts and term definitions differ or overlap slightly, making it difficult to comprehend and connect both formats.
+It also makes the use of a consistent terminology impossible, as some concepts from \gls{xmpp}/\glspl{xep} are not reflected in \gls{sacm} terminology and vice versa.
+
+\subsubsection{Versions of \glspl{xep}}
+\gls{xmpp} Extension Procotols (\glspl{xep}) can receive modifications after being declared final, 
```

Comment:
I don't think that is the main issue since eg. Pubsub is not yet final, and can therefore be modified.
The problem is that most XEPs are "still" drafts (since 10 years and more) and server and clients treating them as de-facto standards without taking change into consideration.
@fabianhauser  agreed?

Reply:
Agreed 👍